plugins/pulp_rpm/plugins/importers/iso/importer.py

```diff
@@ -152,7 +152,10 @@ def upload_unit(self, transfer_repo, type_id, unit_key, metadata, file_path, con
             iso = iso.__class__.objects.get(**iso.unit_key)
 
         # remove any existing units with the same name
+        unit_keys_same_name = models.ISO.objects(name=iso['name']).values_list('id')
         units = repo_controller.find_repo_content_units(transfer_repo.repo_obj,
+                                                        repo_content_unit_q=Q(
+                                                            unit_id__in=unit_keys_same_name),
```

Comment:
@zxiong, if you filter out everything that you need here, does it make sense to filter by name later? See line 159: `units_q=Q(name=iso['name'])`

Reply:
If so, it does not need to filter by name later, so "units_q=Q(name=iso['name']" can be removed.

Reply:
If so, it does not need to filter by name later, so "units_q=Q(name=iso['name']" can be removed.